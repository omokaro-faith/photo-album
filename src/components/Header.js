import React from 'react';
const AlbumPage = () => (
  <header>
    <h1>Faith's PhotoAlbum</h1>
  </header>
)

 export default AlbumPage; 