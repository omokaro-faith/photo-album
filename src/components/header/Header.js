import React from 'react';
import { PropTypes } from 'prop-types';
import '../../styles/components/header/_header.css';

const Header = ({ totalItems, pageName }) => (
  <div className='header__wrapper'>
    <h1 className='header__wrapper--item'>Viewing <span className='header__wrapper--span-item'>{totalItems}</span> {pageName} </h1>
    <h3 className='header__wrapper--item'>Click the navigation buttons to view more {pageName}</h3>
    <h3 className='header__wrapper--item'>Toggle dropdown to get more {pageName}</h3>
  </div>
)

Header.propTypes = {
  totalItems: PropTypes.number.isRequired,
  pageName: PropTypes.string.isRequired,
}



export default Header;
