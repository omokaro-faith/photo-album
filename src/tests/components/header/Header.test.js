import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../../components/header/Header';

describe('Header', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      totalItems: 2,
      pageName: 'Albums'
    };
    wrapper = shallow(<Header {...props}/>);
  });

  afterEach(() => { wrapper = null; });

  
  describe('renders', () => {
    it('should exist', () => {
      expect(wrapper.exists()).toBe(true);
    });

    it('should not have any regressions', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
