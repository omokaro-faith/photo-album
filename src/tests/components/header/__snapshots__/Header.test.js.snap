// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders should not have any regressions 1`] = `ShallowWrapper {}`;
